lib/index.ts

```diff
@@ -1,96 +1,130 @@
+
+  return await buildDepTree(manifestFileContents, lockFileContents, includeDev, strict, manifestFileFullPath);
+}
+
+export async function buildDepTree(
+  manifestFileContents: string,
+  lockFileContents: string,
+  includeDev: boolean = false,
+  strict: boolean = true,
+  defaultManifestFileName: string = 'paket.dependencies',
+): Promise<DepTree> {
+  const manifestFile = parseDependenciesFile(manifestFileContents);
+  const lockFile = parseLockFile(lockFileContents);
+
+  const tree = buildDependencyTree(manifestFile, lockFile, includeDev);
 
-  return parse(manifestFileContents, manifestFileContents, includeDev);
+  tree.name = defaultManifestFileName;
+
+  return tree;
 }
 
-export function buildDependencyTree(
-  /* manifestFile: PaketManifest, */ lockFile: PaketLock, includeDev: boolean = false,
+function buildDependencyTree(
+  manifestFile: PaketDependencies,
+  lockFile: PaketLock,
+  includeDev: boolean = false,
 ) {
   const depTree = {
     dependencies: {},
     name: '',
     version: '',
   } as DepTree;
 
+  const dependencies: { [dep: string]: {} } = {};
+
+  for (const group of manifestFile) {
+    for (const dep of group.dependencies) {
+      const current = dep as NugetDependency;
+      if (current.name) {
+        dependencies[current.name] = {
+          name: current.name,
+        };
+      }
+    }
+  }
+
   for (const group of lockFile.groups) {
-    const isDev = group.name === 'build' || group.name === 'test' || group.name === 'tests';
+    const isDev = ['build', 'test', 'tests'].includes((group.name || '').toLowerCase());
+
     if (isDev && !includeDev) {
       continue;
     }
 
     for (const dep of group.dependencies) {
-      depTree.dependencies[dep.name] = {
-        depType: isDev ? DepType.dev : DepType.prod,
-        dependencies: buildSubTree(dep.dependencies),
-        name: dep.name,
-        version: dep.version,
-      };
+      if (dependencies[dep.name]) {
+        dependencies[dep.name] = {
+          depType: isDev ? DepType.dev : DepType.prod,
+          dependencies: buildSubTree(dep.dependencies, group.dependencies, isDev),
+          name: dep.name,
+          version: dep.version,
+        };
+      }
     }
   }
 
+  depTree.dependencies = dependencies;
+
   return depTree;
 }
 
-function buildSubTree(dependency: any) {
-  const subTree: {[dep: string]: DepTree} = {};
+function buildSubTree(dependencies: any, groupDeps: Dependency[], isDev: boolean) {
+  const subTree: { [dep: string]: {} } = {};
 
-  for (const dep of dependency) {
-    subTree[dep.name] = {
-      dependencies: {},
-      name: dep.name,
-      version: dep.version,
-    };
+  for (const currDep of dependencies) {
+    for (const dep of groupDeps) {
+      if (dep.name === currDep.name) {
+        subTree[dep.name] = {
+          depType: isDev ? DepType.dev : DepType.prod,
+          dependencies: buildSubTree(dep.dependencies, groupDeps, isDev),
+          name: dep.name,
+          version: dep.version,
```

Comment:
I understand that we want to return the object as CLI/plugin expects, but what about the options of the dependency?

Reply:
We have ticket "Apply paket.lock restriction to project's target framework". So, let's think about it later.